packages/kbn-config/src/config.ts

```diff
@@ -31,6 +31,7 @@ export function isConfigPath(value: unknown): value is ConfigPath {
  * @internal
  */
 export interface Config {
+  kbnConfig: any;
```

Comment:
I see the confusion here. Reporting has a strange object that gets passed around from a class called `ReportingConfig`. The reasons we do this go way back, but I believe that passing config around to the parts that need it might be simpler now. 

Given the impact of this, I believe we should add a granular step https://github.com/elastic/kibana/issues/152845: Refactor `ReportingConfig` to just use a simple type. 

@rshen91 Would you mind if we put this PR on pause while we look at that?

Reply:
I think this simplifies the base code quite a lot and hopefully makes things in this PR easier: https://github.com/elastic/kibana/pull/157118.